lib/Client.js

```diff
@@ -733,22 +733,26 @@ class Client extends EventEmitter {
+    * @arg {String} [options.data.components[].custom_id] The ID of the component (type 2 style 0-4, type 3 and type 4 only)
     * @arg {Boolean} [options.data.components[].disabled] Whether the component is disabled (type 2 and 3 only)
     * @arg {Object} [options.data.components[].emoji] The emoji to be displayed in the component (type 2)
-    * @arg {String} [options.data.components[].label] The label to be displayed in the component (type 2)
+    * @arg {String} [options.data.components[].label] The label to be displayed in the component (type 2 and type 4)
     * @arg {Number} [options.data.components[].max_values] The maximum number of items that can be chosen (1-25, default 1)
     * @arg {Number} [options.data.components[].min_values] The minimum number of items that must be chosen (0-25, default 1)
+    * @arg {Number} [options.data.components[].max_length] The maximum input length for a text input, min 1, max 4000 (type 4)
+    * @arg {Number} [options.data.components[].min_length] The minimum input length for a text input, min 0, max 4000 (type 4)
     * @arg {Array<Object>} [options.data.components[].options] The options for this component (type 3 only)
     * @arg {Boolean} [options.data.components[].options[].default] Whether this option should be the default value selected
     * @arg {String} [options.data.components[].options[].description] The description for this option
     * @arg {Object} [options.data.components[].options[].emoji] The emoji to be displayed in this option
     * @arg {String} options.data.components[].options[].label The label for this option
     * @arg {Number | String} options.data.components[].options[].value The value for this option
-    * @arg {String} [options.data.components[].placeholder] The placeholder text for the component when no option is selected (type 3 only)
-    * @arg {Number} [options.data.components[].style] The style of the component (type 2 only) - If 0-4, `custom_id` is required; if 5, `url` is required
+    * @arg {String} [options.data.components[].placeholder] The placeholder text for the component when no option is selected or the text input is empty (type 3 and 4)
+    * @arg {Number} [options.data.components[].style] The style of the component (type 2 and type 4 only) - For buttons, if 0-4, `custom_id` is required; if 5, `url` is required, for text inputs 1-2
     * @arg {Number} options.data.components[].type The type of component - If 1, it is a collection and a `components` array (nested) is required; if 2, it is a button; if 3, it is a select menu
     * @arg {String} [options.data.components[].url] The URL that the component should open for users (type 2 style 5 only)
+    * @arg {Boolean} [options.data.components[].required] Whether this component is required to be filled, default false (type 4)
+    * @arg {String} [options.data.components[].value] A a pre-filled value for this component, max 4000 characters (type 4)
```

Comment:
We should remove this mess and just replace it with a link to Discord's docs, like we've done for `embed`/`embeds` and the like